checkov/gitlab_ci/common/resource_id_utils.py

```diff
@@ -3,16 +3,54 @@
 
 from checkov.common.util.consts import START_LINE, END_LINE
 
+IMAGE_BLOCK_NAMES = ('image', 'services')
+
 
 def generate_resource_key_recursive(conf: dict[str, Any] | list[str] | str, key: str, start_line: int,
                                     end_line: int) -> str:
+    return _generate_resource_key_recursive(conf, key, start_line, end_line, set(), 0)
```

Comment:
Instead of creating another method you can just set the last two arguments as optional params in generate_resource_key_recursive and set their  default value to set() and zero.

Reply:
sounds good, but `scanned_image_blocks` should be `None` as default, otherwise you have a mutable object as default, which will result in some surprises, not good ones 😄